DOC Document changes to trailing slash in URLs (#136)
silverstripe/developer-docs: en/02_Developer_Guides/15_Customising_the_Admin_Interface/02_CMS_Architecture.md

[warning]
Depending on your configuration, the returned value _may or may not_ include a trailing slash. The default is to not include one, but you should take care to not
explicitly expect one scenario or the other.

In PHP you can use [Controller::join_links()](api:SilverStripe\Control\Controller::join_links()) or pass an argument to
[AdminRootController::admin_url()](api:SilverStripe\Admin\AdminRootController::admin_url()) to ensure only one `/` character separates the admin URL from the rest of
your path.

In javascript, if you are using [@silverstripe/webpack-config](https://www.npmjs.com/package/@silverstripe/webpack-config), you can use the `joinUrlPaths()` utility
function.
[/warning]

In PHP you should use:

```php
SilverStripe\Admin\AdminRootController::admin_url()
// This method can take an argument:
SilverStripe\Admin\AdminRootController::admin_url('more/path/here')
```

When writing templates use:

```ss
$AdminURL
<%-- This is actually a method that can take an argument: --%>
$AdminURL('more/path/here')
```

And in JavaScript, this is available through the `ss` namespace

```js
ss.config.adminUrl

> // To add to the path you can use joinUrlPaths from @silverstripe/webpack-config

Reviewer: ```suggestion
// You can use this if you use @silverstripe/webpack-config
```

Reviewer 2: done
